openmls/src/framing/message.rs

```diff
@@ -156,6 +156,18 @@ impl MlsMessageIn {
         self.mls_message.is_handshake_message()
     }
 
+    /// Returns `true` if this is a handshake message and either an external proposal or external commit
+    pub fn is_external(&self) -> bool {
+        let is_external = match &self.mls_message.body {
+            MlsMessageBody::Plaintext(p) => {
+                matches!(p.sender(), Sender::NewMember | Sender::Preconfigured(_))
+            }
+            // external message cannot be encrypted
+            MlsMessageBody::Ciphertext(_) => false,
+        };
+        is_external && self.mls_message.is_handshake_message()
+    }
```

Comment:
I think the check of whether a message is external only depends on its sender type. By design, external messages cannot be encrypted. The library cannot produce them, nor could it decrypt them. I'd simplify this only to check the sender type.